pkg/process/util/address.go

```diff
@@ -134,9 +135,7 @@ func V6AddressFromBytes(buf []byte) Address {
 }
 
 // IPBufferPool is meant to be used in conjunction with `NetIPFromAddress`
-var IPBufferPool = sync.Pool{
-	New: func() interface{} {
-		b := make([]byte, net.IPv6len)
-		return &b
-	},
-}
+var IPBufferPool = ddsync.NewTypedPool(func() *[]byte {
```

Comment:
Did you consider just mimicking the `sync.Pool` init scheme here? Something like 
```go
var IPBufferPool = ddsync.Pool[[]byte]{
  New: func() *[]byte {
    b := make([]byte, net.IPv6len)
    return &b
  },
}
```
You would have to lazy init (using `sync.Once`?) the wrapped pool in `Get()`. You can also lazy-init `New` in `Get()` to the return `new(K)`.

Reply:
I don't want to introduce another mutex+atomic (`sync.Once`) into a potential hot-path of `Get()`.

Reply:
I also don't like the API of `sync.Pool` since `New` is required for all our use cases. I'd rather it be part of a constructor-type function.

Reply:
`sync.Once` is pretty lightweight, so I wouldn't be too worried about performance here. I suggested this since the `sync.Pool` interface is what people are familiar with. But I don't feel too strongly about it.